src/sqlfluff/utils/analysis/select_crawler.py

```diff
@@ -232,6 +232,7 @@ def __init__(
         dialect: Dialect,
         parent: Optional[Query] = None,
         query_class: Type = Query,
+        parent_stack: Optional[Tuple[BaseSegment, ...]] = None,
```

Comment:
This looks good to me. Could you look at adding test(s) for it in `test/utils/analysis/test_select_crawler.py`?

Reply:
@barrywhart Yes! Added a test case for crawling the parent queries